bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -1147,21 +1151,28 @@ private LedgerDirsManager.LedgerDirsListener getLedgerDirsListener() {
 
             @Override
             public void diskAlmostFull(File disk) {
-                if (gcThread.isForceGCAllowWhenNoSpace()) {
-                    gcThread.enableForceGC();
-                } else {
-                    gcThread.suspendMajorGC();
+                if (ledgerDirsManager.getAllLedgerDirs().contains(disk)
+                        || indexDirsManager.getAllLedgerDirs().contains(disk)) {
```

Comment:
We'd better not take indexDirs into consideration. If the index dir shares the same disk with the ledger dir, the LedgerDirsManager can cover the event. If the index dir uses different disks with ledger dir, it doesn't make sense to suspend ledger disk garbage collection.